External application

lisbethmarianne opened the pull request:
Addresses Issue #168 

I added a field application_link to events.
The organizers can fill this field if they want to manage the applications themselves or they can leave this field blank if they want Travis Foundation to manage them (via this app).
The apply link will be the link to the external application form if the organizer put one in.

What do you think? If we want to use this or something like this we would get much more events to use this site (for advertising their diversity tickets). We might have to adjust some texts and other things.

@anikalindtner & @alicetragedy

anikalindtner commented:
I love this and a lot of people have been asking about it already. So I think we should definitely add it! I'm happy to provide the text for it. I think we should then have a separate section for the events that are external.

alicetragedy commented on app/views/events/_form.html.erb:
```diff
@@ -48,6 +48,12 @@
   </div>
 
+  <h3 class="box-subtitle">Manage Application</h3>
+  <p>You can manage the diversity applications for your event by yourself or let Travis Foundation do the work for you. If you wish to manage the applications by yourself, fill in a link below, where applicants will find your application form. If you wish Travis Foundation to mangage the applications for your event, leave this blank.</p>
```
found a minor typo ("mangage" -> "manage")

alicetragedy commented:
I'd also be totally up for this. 👍 
And I'm happy to help where I'm needed when it comes to the implementation, or copywriting, or testing stuff out.

lisbethmarianne commented:
I adapted this PR to @ariaru's changes and put in the text from @anikalindtner. 

I need some help working out the radio buttons in the form. The 3rd one works separated from the first 2. Means you can select the first one and the third one. Which we don't want. What is the best way to solve this? Code see /app/views/events/_form.html.erb.

![bildschirmfoto 2016-06-12 um 14 01 11](https://cloud.githubusercontent.com/assets/11691539/15991046/fe90981e-30a6-11e6-9612-9c656abcc1d2.png)

fidothe commented:
The way Radio buttons in forms work is that they're grouped by name. The reason they're not working here is that the first two have the same name (`:selection_by_organiser`), but the third one has a different name (`:application_by_organizer`). The short-term solution would be to use one name (like `:application_process`) and have a different value for each button (instead of `true` and `false`, probably something like `:selection_by_travis`, `:selection_by_organiser`, and `:application_by_organiser). The longer-term solution is probably to introduce a new object that models this application / selection process stuff. It's probably worth discussing this modelling aspect of it, which might have pretty big implications, tomorrow evening.

lisbethmarianne commented:
Rebased this, and added the application_process (string) to the database that Matt suggested. We can now look at how to add all the validations.

(@ariaru you might need to do a rollback, if you want to continue working on this.)

(This breaks all the tests, sorry, have to check later what is happening.)

lisbethmarianne commented:
Last commit was just to make the tests pass again. 😁 
We still have to do the fancy validations.

lisbethmarianne commented:
Made some test events and it all seems fine to me. So 🎉,  I'm going to merge and deploy this.

@anikalindtner This means you now have 3 options available how events can use diversitytickets.org. 👌
